dev/breeze/pyproject.toml

```diff
@@ -16,5 +16,5 @@
 # under the License.
 [tool.black]
 line-length = 110
-target-version = ['py37', 'py37', 'py38', 'py39']
+target-version = ['py37', 'py38', 'py39', 'py310']
```

Comment:
This is the only change I did, all the other were via pre-commit hooks

Reply:
I figured as much, but this has me worried about thrashing in those screenshots...

Reply:
Fixed, looks like when using GitHub Desktop client to add commit message, some of the pre-commit fails and updates the svg's.

The one that failed was:

- update-breeze-file

Reply:
<img width="1266" alt="image" src="https://user-images.githubusercontent.com/8811558/163046816-8572843a-a7e4-4f3a-b15a-a56ea407d00f.png">

Reply:
You need to have`breeze` installed to update the screenshots - and normally it will only actually run when you modify any of breeze's code (which should basically hapen when you already have breeze installed).

Reply:
(the reason is that the pre-commit **actually runs** breeze to generate the screenshots. They **should** come unchanged if you re-run the pre-commit, but maybe there is a case when they are not?

We also have this one in progress #22908

Reply:
Yeah the screenshots were modified liked I said when using GitHub desktop but worked fine when using CLI.

In both cases, pre-commit hooks were running

Reply:
hmm Static Checks on CI are failing with the following error:

```
Update output of breeze commands in BREEZE.rst...........................................Failed
- hook id: update-breeze-file
- exit code: 1

Traceback (most recent call last):
  File "/home/runner/actions-runner/_work/_tool/Python/3.7.12/x64/bin/breeze", line 33, in <module>
    sys.exit(load_entry_point('apache-airflow-breeze', 'console_scripts', 'breeze')())
  File "/home/runner/actions-runner/_work/_tool/Python/3.7.12/x64/bin/breeze", line 22, in importlib_load_entry_point
    for entry_point in distribution(dist_name).entry_points
  File "/home/runner/.local/lib/python3.7/site-packages/importlib_metadata/__init__.py", line 975, in distribution
    return Distribution.from_name(distribution_name)
  File "/home/runner/.local/lib/python3.7/site-packages/importlib_metadata/__init__.py", line 566, in from_name
    raise PackageNotFoundError(name)
importlib_metadata.PackageNotFoundError: No package metadata was found for apache-airflow-breeze
Traceback (most recent call last):
  File "./scripts/ci/pre_commit/pre_commit_breeze_cmd_line.py", line 81, in <module>
    print_help_for_all_commands()
  File "./scripts/ci/pre_commit/pre_commit_breeze_cmd_line.py", line 46, in print_help_for_all_commands
    check_call(["breeze", "--help"], env=env)
  File "/home/runner/actions-runner/_work/_tool/Python/3.7.12/x64/lib/python3.7/subprocess.py", line 363, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['breeze', '--help']' returned non-zero exit status 1.

```

Reply:
It looks like terminal width is much less than the one that shoudl be enforced when you run it in Github Desktop (see the screenshot). So possibly there is another width involved (I already modify it in 3 places I think but it seems that it can get overridden in a different way.

Reply:
Yep. Reproduced it on my Github Desktop

Reply:
🎉 any ideas on CI failure

Reply:
importlib_metadata.PackageNotFoundError: No package metadata was found for apache-airflow-breeze

Reply:
Likely a but in setup.cfg

Reply:
Interesting. Might be a bug connected with adding Python 3.10 - there were some changes with importlib_metadata in 3.10......

Reply:
BTW. Seems like GitHub Docker for some reasons sets terminal width to 80 :)

Reply:
Let me try and remove Py310 from that list

Reply:
OK. Found the reason for width for now in rich code::

```
        if self.is_dumb_terminal:
            return ConsoleDimensions(80, 25)
```

Reply:
```
    @property
    def is_dumb_terminal(self) -> bool:
        """Detect dumb terminal.

        Returns:
            bool: True if writing to a dumb terminal, otherwise False.

        """
        _term = self._environ.get("TERM", "")
        is_dumb = _term.lower() in ("dumb", "unknown")
        return self.is_terminal and is_dumb
```

Reply:
Static checks still fails even after Removing Py310.

So looks like it unrelated to Py3.7

```
Update output of breeze commands in BREEZE.rst...........................................Failed
- hook id: update-breeze-file
- exit code: 1

Traceback (most recent call last):
  File "/home/runner/actions-runner/_work/_tool/Python/3.7.12/x64/bin/breeze", line 33, in <module>
    sys.exit(load_entry_point('apache-airflow-breeze', 'console_scripts', 'breeze')())
  File "/home/runner/actions-runner/_work/_tool/Python/3.7.12/x64/bin/breeze", line 22, in importlib_load_entry_point
    for entry_point in distribution(dist_name).entry_points
  File "/home/runner/.local/lib/python3.7/site-packages/importlib_metadata/__init__.py", line 975, in distribution
    return Distribution.from_name(distribution_name)
  File "/home/runner/.local/lib/python3.7/site-packages/importlib_metadata/__init__.py", line 566, in from_name
    raise PackageNotFoundError(name)
importlib_metadata.PackageNotFoundError: No package metadata was found for apache-airflow-breeze
Traceback (most recent call last):
  File "./scripts/ci/pre_commit/pre_commit_breeze_cmd_line.py", line 81, in <module>
    print_help_for_all_commands()
  File "./scripts/ci/pre_commit/pre_commit_breeze_cmd_line.py", line 46, in print_help_for_all_commands
    check_call(["breeze", "--help"], env=env)
  File "/home/runner/actions-runner/_work/_tool/Python/3.7.12/x64/lib/python3.7/subprocess.py", line 363, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['breeze', '--help']' returned non-zero exit status 1.

```

Reply:
https://github.com/apache/airflow/pull/22962 in the meantime.

Reply:
Reallly 😕

Reply:
OK. I have a hypothesis why it could happen. Very interesting one.

It is likely the  cache for pre-commit. Pre-commits have not changed, so cache is used from the previous installation 

When you look at the job you will find:

Cache restored from key: pre-commit-local-installation-3.7.12-2efd07d398ec3425dc1a133312e5b3d0ee79da043092c916eb7b39549950cc29

This means that cache for pre-commit venv is used from the cache. And I tihnk it does not contain "breeze" as package because the first time the cache was built - breeze was not installed. We actually did not have breeze installed in "basic-static-checks" job until this mid-day when I merged Bowrna's fixl.

So I think the problem is that you run the first build for the branch (cache is per branch) you did not have latest main and the cache for pre-commit was built then - without breeze. 

So now when the pre-commit cache is restored, breeze package information will be missing. :) 

To test it - make a single character change to `.pre-commit-config.yaml`  - this should change the hash used to identify the cache for pre-commit (pre-commit-local-installation-3.7.12-2efd07d398ec3425dc1a133312e5b3d0ee79da043092c916eb7b39549950cc29. will change) and all should be fine.

Reply:
Hmmm `cache invalidation .....` etc.

Reply:
that must be it, I didn't knew we had that change, today -- let's see

Reply:
Now `pipx` doesn't exist. So looks related

Reply:
Yeah. Now we hit another "teething" problem :)